js/search/search.js

```diff
@@ -178,57 +158,72 @@ class Search {
      * An array of messages to be indexed
      * in real time
      * @param message
+     * @param callback
      */
-    realTimeIndexing(message) {
+    realTimeIndexing(message, callback) {
+        if (typeof callback !== "function") {
+            return false;
+        }
 
         try {
             let msg = JSON.parse(message);
             if (!(msg instanceof Array)) {
-                log.send(logLevels.ERROR, 'RealTime Indexing: Messages must be an array real-time indexing');
-                return (new Error('RealTime Indexing: Messages must be an array'));
+                return callback(false, 'RealTime Indexing: Messages must be an array');
             }
         } catch(e) {
-            log.send(logLevels.ERROR, 'RealTime Indexing: parse error -> ' + e);
-            throw (new Error(e));
+            return callback(false, 'RealTime Indexing: parse error ');
         }
 
         if (!this.isInitialized) {
-            log.send(logLevels.ERROR, 'Library not initialized');
-            throw new Error('Library not initialized');
-        }
-
-        if (!fs.existsSync(searchConfig.FOLDERS_CONSTANTS.INDEX_PATH)) {
-            log.send(logLevels.ERROR, 'User index folder not found');
-            throw new Error('User index folder not found');
+            return callback(false, 'Library not initialized');
         }
 
         this.isRealTimeIndexing = true;
-        return libSymphonySearch.symSEIndexRealTimeAsync(searchConfig.FOLDERS_CONSTANTS.TEMP_REAL_TIME_INDEX, message, (err, result) => {
+        libSymphonySearch.symSEIndexRealtimeRAMAsync(message, (err, result) => {
             this.isRealTimeIndexing = false;
             if (err) {
-                log.send(logLevels.ERROR, 'RealTime Indexing: error -> ' + err);
-                throw new Error(err);
+                return callback(false, 'RealTime Indexing: error');
             }
-            return result;
+            return callback(true, result);
         });
-    }
 
-    /**
-     * Reading a json file
-     * for the demo search app only
-     * @param {String} batch
-     * @returns {Promise}
-     */
-    readJson(batch) {
-        return readFile.call(this, batch);
+        return null;
     }
 
     /**
      * Encrypting the index after the merging the index
      * to the main user index
      */
     encryptIndex(key) {
-        return this.crypto.encryption(key);
+        const mainIndexFolder = path.join(searchConfig.FOLDERS_CONSTANTS.INDEX_PATH,
+            `${searchConfig.FOLDERS_CONSTANTS.PREFIX_NAME}_${this.userId}`);
+        return new Promise(resolve => {
+            if (!fs.existsSync(mainIndexFolder)) {
```

Comment:
What we could do here instead is have a method that could check if the `mainIndexFolder` exists during initialisation and set a bool in this class so that all these methods could use it.

Or, if you believe that a user or any other application/program could arbitrarily remove these folders, then make a check every time we use this method by invoking that abstracted out method.